client/src/ContextProvider/actions.js

```diff
@@ -0,0 +1,56 @@
+// change this if the server address/port changes
+const ROOT_URL = 'http://localhost:3001';
+
+const registerUser = async (dispatch, registerPayload) => {
+  const requestOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      email: registerPayload.registerEmail,
+      name: registerPayload.registerName,
+      password: registerPayload.registerPassword,
+      confirmPassword: registerPayload.confirmPassword,
+    }),
+  };
+
+  try {
+    const response = await fetch(`${ROOT_URL}/api/signup`, requestOptions);
```

Comment:
```suggestion
    const response = await fetch("/api/signup", requestOptions);
```
You shouldn't need the root url here, it will also cause issues with deployment if you do set it here

Reply:
Removed the root url